cosmo_tester/framework/fixtures.py

```diff
@@ -89,18 +89,26 @@ def distributed_installation(cfy, ssh_key, module_tmpdir, attributes, logger,
     If request.param has a 'sanity' value: 6 nodes are built (1 DB 1 Queue
     3 managers and an AIO manager)
 
+    If request.param has a 'cluster_upgrade' value: 8 nodes are built (1 DB 1
+    Queue 3 managers and 3 old AIO managers will be sent multiple times with...
+    ...'cluster_version' testing the cluster upgrade from 4.6 to later versions
+
     With every call 'template_inputs' and 'tf_template' can be passed in when
     creating the TestHosts object
 
     If you need to have your own preconfigure_callback it can be supplied and
     will run before the cluster one
     """
-    cluster, sanity, template_inputs, tf_template = False, False, None, None
+    cluster, sanity, cluster_upgrade, cluster_version = \
+        False, False, False, False
+    template_inputs, tf_template = None, None
     supplied_preconfigure_callback = None
     # request isn't created with a 'param' attribute if no params are sent
     if hasattr(request, 'param'):
         cluster = 'cluster' in request.param
         sanity = 'sanity' in request.param
+        cluster_upgrade = 'cluster_upgrade' in request.param
```

Comment:
maybe something like 'upgrade_cluster_verison'

Reply:
It should be a True or False.
If it's True then a cluster_version is inspected to know which cluster version to upgrade to

Reply:
so maybe 'upgrade_cluser' like in command form

Reply:
Fixed